mlkem/asm/clean/rej_uniform_asm.s

```diff
@@ -0,0 +1,369 @@
+**************************************************/
+.align 4
+.global simd_memcpy
+.global _simd_memcpy
+simd_memcpy:
+_simd_memcpy:
+    /* Input registers */
+    dst                         .req x0
+    src                         .req x1
+    copylen                     .req x2
+
+    /* Temporary registers, no need to preserve */
+    data                        .req w12
+    aligned_addr                .req x13
+    aligned_len                 .req x14
+    unaligned_len               .req x15
+
+    cmp     copylen, #0                     // If copylen is less than or equal to 0
+    b.le    end                             // then we exit the function
+    cmp     copylen, #32                    // Check if size is less than 32 bytes
+    b.lo    small_copy                      // Branch to small copy if size is less than 32
+
+    // Align destination address
+    mov     aligned_addr, dst
+    and     aligned_addr, aligned_addr, #15
+    cbz     aligned_addr, aligned_copy      // If already aligned, jump to aligned copy
+
+    mov     unaligned_len, #16
+    sub     unaligned_len, unaligned_len, aligned_addr
+    cmp     copylen, unaligned_len
+    b.lo    small_copy                      // If size is smaller than alignment padding, use small copy
+    sub     copylen, copylen, unaligned_len
+
+    small_align:
+    ldrb    data, [src], #1
+    strb    data, [dst], #1
+    subs    unaligned_len, unaligned_len, #1
+    b.ne    small_align
+
+aligned_copy:
+    // Main copy loop using 128-bit Neon registers
+    lsr     aligned_len, copylen, #5        // Calculate the number of 32-byte chunks
+    cbz     aligned_len, tail_copy          // If no chunks, jump to tail copy
+
+    copy_loop:
+    ldp     q0, q1, [src], #32
+    stp     q0, q1, [dst], #32
+    subs    aligned_len, aligned_len, #1
+    b.ne    copy_loop
+
+tail_copy:
+    // Copy any remaining bytes less than 32 bytes
+    and     copylen, copylen, #31
+    cbz     copylen, end
+
+small_copy:
+    ldrb    data, [src], #1
+    strb    data, [dst], #1
+    subs    copylen, copylen, #1
+    b.ne    small_copy
+
+end:
+    ret
+
+/*************************************************
+* Name:        rej_uniform_asm
+*
+* Description: Full SIMD lane, run rejection sampling on uniform random bytes to generate
+*              uniform random integers mod q
+*
+* Arguments:   - int16_t *r:          pointer to output buffer
+*              - unsigned int len:    requested number of 16-bit integers
+*                                     (uniform mod q)
+*              - const uint8_t *buf:  pointer to input buffer
+*                                     (assumed to be uniform random bytes)
+*              - unsigned int buflen: length of input buffer in bytes
+*              - unsigned int *consumed: the length of consumed buffer
+* Returns number of sampled 16-bit integers (at most len)
+**************************************************/
+.macro div48    dst, src
```

Comment:
This is not a secret division, is it? Can we not just use `div`?

Reply:
Correct, it's not secret division, using `div` instruction is not optimized in this case, since we divide by constant. It's faster to multiple with inverse, isn't it?

Reply:
I stand corrected, compilers also don't use `div 48` here.

Alternatively, though, one can just use `sub _, _, #48` in the loop, rather than having an explicit iteration counter. But SLOTHY would not yet support that, so let's leave it.

Reply:
Compilers agree with you that the approach above is faster. Still, this is a one-off, so I don't think this is worth optimizing, given that it comes with some cognitive and verification overhead?

Reply:
Okay, I'll use `div` instruction directly. Update soon.

Reply:
done in https://github.com/pq-code-package/mlkem-c-aarch64/pull/47/commits/b42c7a17e06fd23922bdb6a3ac518cfe517a6e59